crates/stacks/src/actor.rs

```diff
@@ -0,0 +1,58 @@
+use std::path::PathBuf;
+
+use kameo::{Actor, actor::ActorRef, message::Message};
+
+use crate::docker::{start_stacks, stop_stacks};
+
+pub struct Worker {
+    pub stacks: bool,
+    pub port: u16,
+    pub config_dir: PathBuf,
+}
+
+pub enum Msg {
+    SetEnabled(bool),
+}
+
+impl Message<Msg> for Worker {
+    type Reply = ();
+
+    async fn handle(
+        &mut self,
+        msg: Msg,
+        _ctx: &mut kameo::message::Context<Self, Self::Reply>,
+    ) -> Self::Reply {
+        match msg {
+            Msg::SetEnabled(enabled) => {
+                self.stacks = enabled;
+                if enabled {
```

Comment:
this should handle spam better. the app seems to be spamming repeated events (which is also a problem that I'll need to check)

but regardless, if we receive this message and `enabled == self.stacks`, then this should do nothing